src/Analyzer.Cli.UnitTests/CommandLineParserTests.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Azure.Templates.Analyzer.Cli;
+using System.Linq;
+using System;
+using System.IO;
+
+namespace Analyzer.Cli.FunctionalTests
+{
+    [TestClass]
+    public class CommandLineParserTests
+    {
+        private CommandLineParser _commandLineParser;
+
+        [TestInitialize]
+        public void TestInit()
+        {
+            _commandLineParser = new CommandLineParser();
+        }
+
+        [DataTestMethod]
+        [DataRow("Path does not exist", 2, DisplayName = "Invalid template file path provided")]
+        [DataRow("\\CommandLineParserTests.cs", 4, DisplayName = "Path exists, not an ARM template.")]
+        [DataRow("\\CommandLineParserTests.cs", 3, "--report-format", "Sarif", DisplayName = "Path exists, Report-format flag set, --output-file-path flag not included.")]
+        [DataRow("\\CommandLineParserTests.cs", 1, "--parameters-file-path", DisplayName = "Path exists, Parameters-file-path flag included, but no value provided.")]
+        [DataRow("\\AppServicesLogs-Failures.json", 5, DisplayName = "Issues found in the template")]
+        [DataRow("\\AppServicesLogs-Passes.json", 0, DisplayName = "Success")]
+        public void AnalyzeTemplate_ValidInputValues_ReturnExpectedExitCode(string relativeTemplatePath, int expectedExitCode, params string[] additionalParams)
+        {
+            var args = new string[] { "analyze-template" , String.Concat(Directory.GetCurrentDirectory(), relativeTemplatePath)}; 
+            args = args.Concat(additionalParams).ToArray();
+            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
+
+            Assert.AreEqual(expectedExitCode, result.Result);
+        }
+
+        [DataTestMethod]
+        [DataRow("\\Configuration.json", 1, DisplayName = "Provided parameters file is not a parameters file")]
+        [DataRow("\\Parameters.json", 5, DisplayName = "Provided parameters file correct, issues in template")]
+        public void AnalyzeTemplate_ParameterFileParamUsed_ReturnExpectedExitCode(string relativeParametersFilePath, int expectedExitCode)
+        {
+            var templatePath = String.Concat(Directory.GetCurrentDirectory(), "\\AppServicesLogs-Failures.json");
+            var parametersFilePath = String.Concat(Directory.GetCurrentDirectory(), relativeParametersFilePath);
+            var args = new string[] { "analyze-template", templatePath, "--parameters-file-path", parametersFilePath };
+            var result = _commandLineParser.InvokeCommandLineAPIAsync(args);
+
+            Assert.AreEqual(expectedExitCode, result.Result);
+        }
+
+        [TestMethod]
+        public void AnalyzeTemplate_ConfigurationFileParamUsed_ReturnExpectedExitCode()
```

Comment:
It's not clear to me what this scenario is actually testing. Can you create a more descriptive name? (The CLI will always return some kind of exit code. What is special about it when using a configuration file? Will it always be the same when using a configuration file?)  Same with the Sarif test variant below.